### PR TITLE
feat: image compression

### PR DESCRIPTION
**Pull request by artegoser**
Adds image compressing proxy

closes: #96

### CODE REVIEW

File: src/routes/browser/proxy.ts [hunk 1]
```diff
@@ -17,4 +20,49 @@ export default async function proxyRoute(fastify: FastifyInstance) {
...
+            quality: 25,
+            progressive: true,
+            optimizeScans: true,
+          })
+          .toBuffer();
+
+        reply.header('Content-Type', 'image/webp');
+        reply.header('Content-Length', buffer.length);
+
+        reply.header('x-original-size', clen);
```
Comment (DarkCat09) [inline src/routes/browser/proxy.ts:62]: Do we need this? Really?

File: src/routes/browser/proxy.ts [hunk 2]
```diff
@@ -17,4 +20,49 @@ export default async function proxyRoute(fastify: FastifyInstance) {
...
+      { schema: ProxySchema },
+      async (request, reply) => {
+        const response = await axios.get(request.query.url, {
+          responseType: 'arraybuffer',
+        });
+
+        const mime: string | undefined =
+          response.headers['content-type']?.toString();
+
+        if (!(mime && mime.startsWith('image'))) {
```
Comment (DarkCat09) [inline src/routes/browser/proxy.ts:36]: `image/`, not `image`

File: src/utils/generate.ts [hunk 3]
```diff
@@ -27,12 +27,13 @@ export function generateParserUrl(
...
   remoteUrl: URL,
-  href: string
+  href: string,
+  subProxy?: string
 ): string {
   const realHref = getRealURL(href, remoteUrl);
 
   const urlParam = `?url=${encodeURIComponent(realHref.href)}`;
-  return `${requestUrl.origin}/proxy${urlParam}`;
+  return `${requestUrl.origin}/proxy${subProxy ? `/${subProxy}` : ''}${urlParam}`;
```
Comment (DarkCat09) [inline src/utils/generate.ts:36]: Unreadable...  
I think we should have implemented MIME type restriction first, #98

Comment (artegoser) [reply src/utils/generate.ts:36]: Maybe accept the pull request first, and then fix it under mime type restriction?

**Review by DarkCat09 — APPROVED** [review]

**Pull request merged**